pkg/kube/watcher.go

```diff
@@ -67,29 +70,56 @@ func (e *EventWatcher) OnUpdate(oldObj, newObj interface{}) {
 	e.onEvent(event)
 }
 
-// Ignore events older than the maxEventAgeSeconds
+// Ignore events older than the maxEventAgeSeconds.
 func (e *EventWatcher) isEventDiscarded(event *corev1.Event) bool {
-	timestamp := event.LastTimestamp.Time
-	if timestamp.IsZero() {
-		timestamp = event.EventTime.Time
-	}
-	eventAge := time.Since(timestamp)
+	eventAge, timeUsedForAge := getEventAge(event, clock.RealClock{})
 	if eventAge > e.maxEventAgeSeconds {
 		// Log discarded events if they were created after the watcher started
-		// (to suppres warnings from initial synchrnization)
-		if timestamp.After(startUpTime) {
+		// (to suppres warnings from initial synchronization)
+		if timeUsedForAge.After(startUpTime) {
 			log.Warn().
-				Str("event age", eventAge.String()).
-				Str("event namespace", event.Namespace).
-				Str("event name", event.Name).
-				Msg("Event discarded as being older then maxEventAgeSeconds")
+				Str("eventAge", eventAge.String()).
```

Comment:
Nit-- Two suggestions: (1) capture `currentTime` separately, pass it to `getEventAge` and (2) include both `currentTime` and `timeUsedForAge` in the warning.